tests/UrlTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * This file is part of the webmozart/path-util package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Webmozart\PathUtil\Tests;
+
+use Webmozart\PathUtil\Url;
+
+/**
+ * @since  2.3
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Claudio Zizza <claudio@budgegeria.de>
+ */
+class UrlTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider provideMakeRelativeTests
+     * @covers Webmozart\PathUtil\Url
+     */
+    public function testMakeRelative($absolutePath, $basePath, $relativePath)
+    {
+        $host = 'http://example.com';
+
+        $relative = Url::makeRelative($host.$absolutePath, $host.$basePath);
+        $this->assertSame($relativePath, $relative);
+    }
+
+    /**
+     * @dataProvider provideMakeRelativeTests
+     * @covers Webmozart\PathUtil\Url
+     */
+    public function testMakeRelativeWithFullUrl($absolutePath, $basePath, $relativePath)
+    {
+        $host = 'ftp://user:password@example.com:8080';
+
+        $relative = Url::makeRelative($host.$absolutePath, $host.$basePath);
+        $this->assertSame($relativePath, $relative);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage "/webmozart/puli/css/style.css" is not a valid Url.
+     * @covers Webmozart\PathUtil\Url
+     */
+    public function testMakeRelativeWithoutFullUrl()
+    {
+        $relative = Url::makeRelative('/webmozart/puli/css/style.css', 'http://example.com/webmozart/puli');
```

Comment:
@webmozart Could you please take a look on the new changes?

I decided to not simply return a relative path because of paths like `css/../style.css` should be normalized and I don't want to use another method from `Path` in `Url::makeRelative` to prevent another static dependency. `Path::makeRelative` is handling this internally.